cmd/generate/config/rules/snyk.go

```diff
@@ -0,0 +1,22 @@
+package rules
+
+import (
+	"github.com/zricethezav/gitleaks/v8/config"
+)
+
+func Snyk() *config.Rule {
+	// define rule
+	r := config.Rule{
+		Description: "Snyk API token",
+		RuleID:      "snyk-api-token",
+		SecretGroup: 1,
+		Regex:       generateSemiGenericRegex([]string{"snyk_token"}, hex8_4_4_4_12()),
```

Comment:
@wayne-snyk I would argue this could be reduced to just `snyk`, WDYT? For example, we'd still want to match on strings like `const SNYK_KEY="12345678-ABCD-ABCD-ABCD-1234567890AB"`.